libraries/azure-app-configuration-importer/src/internal/parsers/defaultConfigurationSettingsConverter.ts

```diff
@@ -32,43 +34,68 @@ export class DefaultConfigurationSettingsConverter implements ConfigurationSetti
   public Convert(
     config: object,
     options: SourceOptions
-  ): SetConfigurationSettingParam<string | FeatureFlagValue | SecretReferenceValue>[] {
+  ): SetConfigurationSettingParam<string | MsFeatureFlagValue | SecretReferenceValue>[] {
     let configurationSettings = new Array<
-      SetConfigurationSettingParam<string | FeatureFlagValue>
+      SetConfigurationSettingParam<string | MsFeatureFlagValue>
     >();
-    let featureFlagsConfigSettings = new Array<SetConfigurationSettingParam<FeatureFlagValue>>();
-    let foundFeatureManagement = false;
-    let featureFlags: any = {};
-    let featureManagementKeyWord = "";
-    let enabledForKeyWord = "";
+
+    let featureFlagsConfigSettings = new Array<SetConfigurationSettingParam<MsFeatureFlagValue>>();
+    let foundMsFmSchema = false;
+    let foundLegacySchema = false;
+    let legacySchemaFeatureManagementKeyWord = "";
+
+    const featureFlagsDict: any = {};
 
     if (options.format == ConfigurationFormat.Properties && !options.skipFeatureFlags) {
       this.checkFeatureManagementExist(config);
     }
 
-    for (let i = 0; i < Constants.FeatureManagementKeyWords.length; i++) {
+    for (let i = 0; i < Constants.FeatureManagementKeyWords.length - 1; i++) {
       if (Constants.FeatureManagementKeyWords[i] in config) {
-        if (foundFeatureManagement) {
+        if (foundLegacySchema) {
           throw new ArgumentError(
-            "Unable to proceed because data contains multiple sections corresponding to Feature Management."
+            `Unable to proceed because data contains multiple sections corresponding to Feature Management. with the key, ${Constants.FeatureManagementKeyWords[i]}`
           );
         }
+        
+        foundLegacySchema = true;
+        legacySchemaFeatureManagementKeyWord = Constants.FeatureManagementKeyWords[i];
+
+        const legacyFeatureManagementKey = legacySchemaFeatureManagementKeyWord as keyof object;
+        featureFlagsDict[legacyFeatureManagementKey] = config[legacyFeatureManagementKey];
+        delete config[legacyFeatureManagementKey];
+      }
+    }
+
+    const msFeatureManagementKeyWord = Constants.FeatureManagementKeyWords[3];
+    if (msFeatureManagementKeyWord in config) {
+      if (foundLegacySchema &&
+        Object.keys(config[msFeatureManagementKeyWord as keyof object]).some(key => key !== Constants.FeatureFlagsKeyWord)) {
+        throw new ArgumentError(
+          `Unable to proceed because data contains an already defined legacy schema section with the key, ${legacySchemaFeatureManagementKeyWord}.`
+        );
+      }
+
+      if (Object.keys(config[msFeatureManagementKeyWord as keyof object]).includes(Constants.FeatureFlagsKeyWord)) {
+        foundMsFmSchema = true;
+      }
```

Comment:
Would this also work ?
```
 const msFeatureManagementKeyWord = Constants.FeatureManagementKeyWords[3];
 if (Object.keys(config[msFeatureManagementKeyWord as keyof object]).includes(Constants.FeatureFlagsKeyWord)) {
        foundMsFmSchema = true;
      }

    if (msFeatureManagementKeyWord in config) {
      if (foundLegacySchema && !foundMsFmSchema)) {
        throw new ArgumentError(
          `Unable to proceed because data contains an already defined legacy schema section with the key, ${legacySchemaFeatureManagementKeyWord}.`
        );
      }

     
```

Reply:
Don't we need to check if `msFeatureManagementKeyword` exists first before this line:
`if (Object.keys(config[msFeatureManagementKeyWord as keyof object]).includes(Constants.FeatureFlagsKeyWord)) {
        foundMsFmSchema = true;
      }`

Reply:
My bad we do I missed adding the initial bracket this is what I meant

```
 if (msFeatureManagementKeyWord in config) {
     if (Object.keys(config[msFeatureManagementKeyWord as keyof object]).includes(Constants.FeatureFlagsKeyWord)) {
        foundMsFmSchema = true;
      
      if (foundLegacySchema && !foundMsFmSchema) {
        throw new ArgumentError(
          `Unable to proceed because data contains an already defined legacy schema section with the key, ${legacySchemaFeatureManagementKeyWord}.`
        );
      }

     }
     
```

Reply:
For the case you shared above wouldn't this code 
```
 if (msFeatureManagementKeyWord in config) {
      if (foundDotnetFmSchema &&
        Object.keys(config[msFeatureManagementKeyWord as keyof object]).some(key => key !== Constants.FeatureFlagsKeyWord)) {
        throw new ArgumentError(
          `Unable to proceed because data contains an already defined dotnet schema section with the key, ${dotnetFmSchemaKeyWord}.`
        );
      }
...
```
Also throw for the example you've shared above

Reply:
foundDotnetFmSchema will be true but `Object.keys(config[msFeatureManagementKeyWord as keyof object]).some(key => key !== Constants.FeatureFlagsKeyWord`  this part will return false. It won't throw